### PR TITLE
Add namespace support to service discovery

### PR DESCRIPTION
**Pull request by pgressa**

### CODE REVIEW

**Comment by pgressa** [comment]
This PR is missing the configuration of k8s env, that will be discussed with @alvarosanchez tmrw.

**Comment by pgressa** [comment]
@alvarosanchez please check

File: setup-test-namespace.sh [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+kubectl create secret generic test-secret --from-literal=username='my-app' --from-literal=password='39528$vdg7Jb'
+kubectl create secret generic another-secret --from-literal=secretProperty='secretValue'
+kubectl create secret generic mounted-secret --from-literal=mountedVolumeKey='mountedVolumeValue'
+
+kubectl label configmap game-config-yml app=game
+kubectl label configmap literal-config app=game
+kubectl label secret another-secret app=game
+
+# Create and expose deployment - example service
+cat <<EOF | kubectl apply -f -
```
Comment (alvarosanchez) [inline setup-test-namespace.sh:43]: Could these YAML snippets be stored on their own YAML files?

**Review by alvarosanchez — APPROVED** [review]
Left one comment. Other than that, this is good to go.

File: setup-test-namespace.sh (hunk 1 above)
Comment (pgressa) [reply setup-test-namespace.sh:43]: ok

**Comment by pgressa** [comment]
@alvarosanchez addressed your comment, but seems like the test is flaky a bit, checkout https://github.com/micronaut-projects/micronaut-kubernetes/pull/191/checks?check_run_id=1248388611#step:7:101 , in other checks from that run it passes. When I checked the previous ones, this exact test fails there as well.

File: kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/client/v1/KubernetesServiceConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.EachProperty;
+import io.micronaut.context.annotation.Parameter;
+
+import javax.inject.Inject;
+import java.util.Optional;
+
+/**
+ * Discovery configuration for Kubernetes service.
```
Comment (alvarosanchez) [inline kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/client/v1/KubernetesServiceConfiguration.java:26]: Needs `@since` tag

**Review by alvarosanchez — APPROVED** [review]

**Pull request merged**